lessons/projects/5.mdx

```diff
@@ -0,0 +1,1379 @@
+    functionName: "tokenURI",
+    args: tokenData,
+    watch: true,
+  })
+
+  // useEffect to check user connection...
+
+  useEffect(() => {
+    try {
+      if (tokenURI) {
+        setLatestNFTMinted(JSON.parse(window.atob(tokenURI.substring(tokenURI.indexOf(',') + 1))));
+      }
+    } catch (error) {
+      console.log("Error fetching token URI", error.message);
+    }
+  }, [tokenData, tokenURI]);
+
+  const mintToken = async (e) => {
+    try {
+      // mint txn function
+
+      await mintTxn.wait();
+      console.log("This is the mint data", mintData);
+      refetchTokenData();
+      
+      // catch condition for mintToken...
+  };
+
+  // return statement...
+```
+
+- The first thing we need is the `tokenId` of the latest NFT minted. We 
+can do so by reading the value of the `totalSupply` stored in our smart 
+contract because the `totalSupply` should equal to the latest `tokenId` in
+our case. We can fetch this using the `useContractRead` hook from WAGMI 
+that is configured for reading data from smart contracts. The 
+`refetchTokenData` function will be called in our `mintToken` function 
+to update the `tokenData` after minting has succeeded. We pass a 
+`watch: true` key-value pair so that the hook keeps and active watch on 
+the value of totalSupply.
+- Then the returned value (tokenData) is passed in as an argument to the 
+`tokenURI` function from our smart contract and we received a `Base64` 
+encoded string in return.
+- A `useEffect` hook keeps updating this string everytime the `tokenData` 
+or `tokenURI` is updated and stores a decoded JSON format of it in the 
+`latestNFTMinted` variable.
+
+We're almost there. We just need to do two more checks before we actually
+look at our modal. The first is that we must display our modal only when 
+we want it to and the other is to display a loading message until our NFT 
+has successfully minted. 
+
+```jsx 
+// useEffect for tokenURI...
+
+  const mintToken = async (e) => {
+    try {
+      setIsMinting(true);
+      setModalShow(true);
+      let mintTxn = await mint({
+        recklesslySetUnpreparedOverrides: {
+          value: ethers.utils.parseEther(e.target.value),
+        }
+      });
+      await mintTxn.wait();
+      console.log("This is the mint data", mintData);
+      refetchTokenData();
+      setIsMinting(false);
+    } catch (error) {
+      console.log("Error minting NFT", error.message);
+    }
+  };
+
+// return statement...
+```
+
+When we trigger the `mintToken` function we want to set the state to 
+`isMinting`. Based on this condition, we can inform the users that the 
+mint is in progress. We also want our modal to pop up right after this 
+so we set the value of `modalShow` to `true`.
+
+We're there at last! Now we can render our modal.
+
+```jsx
+// return statement for rendering...
+
+      {isUserConnected ?
+        <main className={styles.main}>
+          <div style={NFTFlex}>
+
+            {/* NFT Card code... */}
```

Comment:
use opening a closing tags of the code that should be in here.
At this stage, I'm cheating. I'm going for the final code at the end.
Sorry @ropats16 !!